### PR TITLE
Add gRPC topic producer endpoint - Java side

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java [hunk 1]
```diff
@@ -67,6 +70,12 @@ abstract class AbstractGrpcAgent extends AbstractAgentCode {
     protected final AtomicBoolean restarting = new AtomicBoolean(false);
 
     @Getter protected volatile boolean startFailedButDevelopmentMode = false;
+    protected AgentServiceGrpc.AgentServiceStub asyncStub;
+
+    AtomicReference<StreamObserver<TopicProducerWriteResult>> topicProducerWriteResults =
```
Comment (eolivelli) [inline langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java:75]: nit: private ?

File: langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java [hunk 2]
```diff
@@ -92,6 +101,68 @@ public void start() throws Exception {
...
+                        new StreamObserver<>() {
+                            @Override
+                            public void onNext(TopicProducerRecord topicProducerRecord) {
+                                TopicProducer topicProducer =
+                                        topicProducers.computeIfAbsent(
+                                                topicProducerRecord.getTopic(),
+                                                topic ->
+                                                        agentContext
+                                                                .getTopicConnectionProvider()
+                                                                .createProducer(
```
Comment (eolivelli) [inline langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java:117]: we also have to "start" the producer

File: langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java [hunk 3]
```diff
@@ -128,8 +199,13 @@ public void stopChannel(boolean wait) throws Exception {
 
     @Override
     public synchronized void close() throws Exception {
+        topicProducerWriteResults.get().onCompleted();
```
Comment (eolivelli) [inline langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java:202]: isn't this leading to ?

```
  agentContext.criticalFailure(
                                        new RuntimeException("Unexpected completion"));
```

Comment (cbornet) [reply langstream-agents/langstream-agent-grpc/src/main/java/ai/langstream/agents/grpc/AbstractGrpcAgent.java:202]: Good catch!

**Review by eolivelli — APPROVED** [review]
LGTM

but a test is failing
```

Error:  Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 15.424 s <<< FAILURE! - in ai.langstream.agents.grpc.AbstractGrpcAgentTest
Error:  ai.langstream.agents.grpc.AbstractGrpcAgentTest.testTopicProducerError  Time elapsed: 15.006 s  <<< ERROR!
java.util.concurrent.TimeoutException
	at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
	at ai.langstream.agents.grpc.AbstractGrpcAgentTest.testTopicProducerError(AbstractGrpcAgentTest.java:177)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

**Pull request merged**